dataclasses_avroschema/schema_generator.py

```diff
@@ -115,30 +115,31 @@ def _get_enum_type_map(cls: Type[CT]) -> Dict[str, enum.EnumMeta]:
         return enum_types
 
     @classmethod
-    def _deserialize_list_type(cls: Type[CT], *, field_name: str, payload: Any) -> List:
-        data: List = []
-        for value in payload:
-            if isinstance(value, dict):
-                data.append(cls._deserialize_complex_types(value))
-            elif isinstance(value, list):
-                data.append(cls._deserialize_list_type(field_name=field_name, payload=value))
-            else:
-                data.append(value)
-        return data
+    def _deserialize_list(cls: Type[CT], field: str, payload: Any) -> List:
+        field_def = cls.schema_def.fields_map[field]
+        item_type = field_def.type.__args__[0]  # it could be List[SOMETHING] or Tuple[Something]
+        forward_type = Type[ForwardRef(field_def.parent.__name__)]
+
+        if item_type == forward_type:  # recurcive case
+            return [field_def.parent._deserialize_complex_types(v) for v in payload]
+        elif issubclass(item_type, AvroModel):
+            return [field_def.type.__args__[0]._deserialize_complex_types(v) for v in payload]
+        else:
+            return [v for v in payload]
 
     @classmethod
     def _deserialize_complex_types(cls: Type[CT], payload: Dict[str, Any]) -> Dict:
         output: Dict[str, Any] = {}
         enum_type_map = cls._get_enum_type_map()
         for field, value in payload.items():
-            if isinstance(value, dict):
```

Comment:
In the current implementation we don't use recursive approach to deserialize nested  objects => cls.schema_def would be always the root DataClass => it's not possible to parse nested object on the third level.